test/s3_test.py

```diff
@@ -83,16 +89,16 @@ def test_read(self):
         t = S3Target('s3://mybucket/tempfile', client=client)
         read_file = t.open()
         file_str = read_file.read()
-        self.assertEqual(self.tempFileContents, file_str)
+        self.assertEqual(self.tempFileContents, file_str.encode('utf8'))
```

Comment:
Nitpick: Actually, utf8 is not a valid identifier. Python has an "utf8" alias for "utf-8", but I think it's a mistake in Python's part. I wouldn't use "utf8" in new code.